### PR TITLE
Add asyncio support

### PR DESCRIPTION
**Pull request by christiansandberg**
Adds support for developing [asyncio](https://docs.python.org/3/library/asyncio.html) applications with python-can more easily. This can be useful when implementing protocols that handles simultaneous connections to many nodes since you can write synchronous looking code without handling multiple threads and locking mechanisms.

For information I started working a year ago or so on adding native asyncio support for each interface but abandoned it since not all API's support callbacks and the performance would probably be worse than using a thread for reading messages.

### CODE REVIEW

**Comment by felixdivo** [comment]
Uhh, looks really nice!

**Comment by felixdivo** [comment]
Could you add a test for this on `Python 3.4+`? This could basically test the same behavior as the normal `BufferedReader` to cover both, right? Maybe in `python-can/test/listener_test.py`.

**Comment by christiansandberg** [comment]
The problem is that testing this will require statements that are not available in Python 2.7 so it won’t even be able to read the file. Not sure if that can be handled by pytest.

**Comment by felixdivo** [comment]
Hm, I see.

File: test/notifier_test.py [hunk 1]
```diff
@@ -16,6 +20,7 @@ def test_single_bus(self):
         bus.send(msg)
         self.assertIsNotNone(reader.get_message(1))
         notifier.stop()
+        bus.shutdown()
```
Comment (felixdivo) [inline test/notifier_test.py]: Just a note: These could also use the bus as a context manager.

Comment (christiansandberg) [reply test/notifier_test.py]: True, but maybe we need to make Notifier a context manager too, otherwise one ends up needing to use try: except: anyway.

**Pull request merged**